library/src/test/kotlin/com/chuckerteam/chucker/internal/support/RequestProcessorTest.kt

```diff
@@ -0,0 +1,75 @@
+package com.chuckerteam.chucker.internal.support
+
+import android.content.Context
+import com.chuckerteam.chucker.api.BodyDecoder
+import com.chuckerteam.chucker.api.ChuckerCollector
+import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
+import io.mockk.every
+import io.mockk.mockk
+import okhttp3.Headers
+import okhttp3.HttpUrl
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+
+internal class RequestProcessorTest {
+    private val context: Context = mockk()
+    private val chuckerCollector: ChuckerCollector = mockk(relaxed = true)
+    private val maxContentLength: Long = 0
+    private val headersToRedact: Set<String> = emptySet()
+    private val bodyDecoders: List<BodyDecoder> = emptyList()
+
+    private val blankGraphQLPath = "----"
+    private val graphQLPath = "graphql"
+
+    private val requestProcessor: RequestProcessor = RequestProcessor(
+        context = context,
+        collector = chuckerCollector,
+        maxContentLength = maxContentLength,
+        headersToRedact = headersToRedact,
+        bodyDecoders = bodyDecoders,
+    )
+
+
+    @Test
+    fun `Given a GraphQL Url path WHEN process request THEN transaction is GraphQLRequest`() {
+        val transaction = HttpTransaction()
+        val graphQLUrl = getGraphQLUrl(graphQLPath)
+
+        val request: Request = mockk(relaxed = true) {
+            every{ url } returns graphQLUrl
+        }
+        requestProcessor.process(request,transaction,graphQLPath)
+        assertTrue(transaction.isGraphQLRequest)
+
+    }
+
+    @Test
+    fun `Given an Url with no GraphQL path WHEN process request THEN transaction is NOT GraphQLRequest`() {
+        val transaction = HttpTransaction()
+        val urlWithNoGraphQLPath = getNonGraphQLUrl()
+        val request:Request = mockk(relaxed = true) {
+            every { url } returns urlWithNoGraphQLPath
+        }
+
+        requestProcessor.process(request,transaction,blankGraphQLPath)
+        assertFalse(transaction.isGraphQLRequest)
+
+    }
+
+    private fun getGraphQLUrl(graphQLPath: String) =  HttpUrl.Builder()
+        .scheme("https")
+        .host("random_host")
+        .addPathSegment(graphQLPath)
+        .build()
+
+    private fun getNonGraphQLUrl() = HttpUrl.Builder()
+        .scheme("https")
+        .host("random_host")
+        .addPathSegment("some/other/path")
+        .build()
+
```

Comment:
nit: since this is all the same except for one parameter, which could be named in a self-documenting way as well, maybe make this one function?